arch/arm/core/sys_fatal_error_handler.c

```diff
@@ -67,7 +67,8 @@ void __weak z_SysFatalErrorHandler(unsigned int reason,
 #endif
 
 	for (;;) {
-		k_cpu_idle();
+		/* Don't do k_cpu_idle here because it will stop the watchdog
```

Comment:
The argument is that calling `k_cpu_idle()` seems to set the CPU in a form of sleep state, which really isn't what a panic should do, or at least not what I'm expecting it to do. I'd expect a panic to busy wait forever. This is how it was done previously in Zephyr as well up until cdb94d64252bf2b4382532932441b13dd91aba6d and

```
-		for (;;)
-			; /* spin forever */
+		goto hang_system;
```

but that commit doesn't really contain any justification for idling rather than busy waiting.

Reply:
Let's get @andrewboie 's view, as the author of this, quite old, change.

Reply:
it's so if the the device is battery powered, it doesn't drain as much.

> which really isn't what a panic should do, or at least not what I'm expecting it to do.

You can't expect any functionality from the system once the kernel has decided to panic.

seems to me that you have special needs for error handling. z_SysFatalErrorHandler() is intentionally declared __weak, for your application just override it and do whatever you like.